plugins-mu/hm-stack/class-endpoint-controller.php

```diff
@@ -0,0 +1,238 @@
+
+		// Fetch all activity/alerts against this site.
+		register_rest_route( $namespace, 'activity', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_site_activity' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+
+		// Fetch bandwidth usage for this site.
+		register_rest_route( $namespace, 'bandwidth-usage', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_bandwidth' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+
+		// Fetch all site environmental data for this site.
+		register_rest_route( $namespace, 'environment-data', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_environment' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+			'schema'              => [
+				'$schema'              => 'http://json-schema.org/draft-04/schema#',
+				'title'                => 'environment-data',
+				'type'                 => 'object',
+				'properties'           => [
+					'environmentData' => [
+						'description'  => esc_html__( 'Data about the current server environment', 'my-textdomain' ),
+						'type'         => 'object',
+						'context'      => array( 'view' ),
+						'readonly'     => true,
+					],
+					'gitData' => [
+						'description'  => esc_html__( 'Data about the current state of git on the site.', 'my-textdomain' ),
+						'type'         => 'object',
+						'context'      => array( 'view' ),
+						'readonly'     => true,
+					],
+				],
+			]
+		] );
+
+		// Fetch all pull requests against this site.
+		register_rest_route( $namespace, 'pull-requests', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_pull_requests' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+
+		// Fetch average page generation time for this site.
+		register_rest_route( $namespace, 'page-generation', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_page_generation_time' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+	}
+
+	/**
+	 * Validate whether the current user has appropriate capabilities to fetch HM Stack data or not.
+	 *
+	 * @return bool
+	 */
+	public function permissions_check() : bool {
+		return current_user_can( 'manage_options' );
+	}
+
+	/**
+	 * Make a human-readable error code from various Hm Stack responses.
+	 *
+	 * @param $error WP_Error Error to parse.
+	 * @return WP_Error
+	 */
+	private function get_return_error_message( WP_Error $error ) : WP_Error {
```

Comment:
This is a bit of a weird name, `get_wp_error_for_hm_stack_response_error`? if that's what it is